src/RecyclableMemoryStreamManager.cs

```diff
@@ -372,24 +342,22 @@ internal byte[] GetLargeBuffer(long requiredSize, string tag)
             else
             {
                 // Buffer is too large to pool. They get a new buffer.
-
+                
                 // We still want to track the size, though, and we've reserved a slot
                 // in the end of the inuse array for nonpooled bytes in use.
                 poolIndex = this.largeBufferInUseSize.Length - 1;
 
                 // We still want to round up to reduce heap fragmentation.
                 buffer = new byte[requiredSize];
-                string callStack = null;
                 if (this.GenerateCallStacks)
                 {
                     // Grab the stack -- we want to know who requires such large buffers
                     callStack = Environment.StackTrace;
-                }
-                Events.Writer.MemoryStreamNonPooledLargeBufferCreated(requiredSize, tag, callStack);
-                ReportLargeBufferCreated();
+                }                
```

Comment:
This is a tricky one, but, even if `GenerateCallStacks` is true, this might be a pooled buffer or the events aren't being subscribed.

It would be better to instantiate a `StackTrace` instance with the necessary `skipFrames`.

Doesn't `EventSource` provide stack trace already?